src/zero_copy/sliced.rs

```diff
@@ -0,0 +1,141 @@
+use core::marker::PhantomData;
+
+use super::{
+    error::Error,
+    input::{Input, InputRef, SliceInput},
+    internal::{Check, Emit, Mode},
+    Located, PResult, Parser,
+};
+
+pub struct RepeatedSlice<A, I: ?Sized, E = (), S = ()> {
+    pub(crate) parser: A,
+    pub(crate) at_least: usize,
+    pub(crate) at_most: Option<usize>,
+    pub(crate) phantom: PhantomData<(E, S, I)>,
+}
```

Comment:
Out of interest, how does this combinator work? Is it akin to `x.repeated().map_with_slice(|_, s| s)`?